sstables/mx/writer.cc

```diff
@@ -574,7 +574,8 @@ class writer : public sstable_writer::writer_impl {
         bytes_ostream blocks; // Serialized pi_blocks.
         bytes_ostream offsets; // Serialized block offsets (uint32_t) relative to the start of "blocks".
```

Comment:
> Before this patch, for "ms" sstables, `to_deletion_time` would have
> be called twice for each written partition tombstone, which would fail
> the test.

Moral: keep pure functions pure